### PR TITLE
Delete item from list

### PR DESCRIPTION
**Pull request by shirleytang0121**
### Related Issue
#7 

### Overview of PR
- [x] add 'deleteItem' router to allow user delete the item from the list

### Technical Changes

- added `deleteItem.js` under `/src/app/list`

### Architecture Changes
- added {{parashop}}/list/deleteitem

### CODE REVIEW

File: src/app/deleteItem/deleteItem.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+export default async function deleteItem(req, res) {
+  try {
+    const { body } = req;
+    const { groceryID } = body;
+    const response = await deleteGrocery({ groceryID });
+    return res.status(httpStatus.OK).jsend.success(response);
+  } catch (err) {
+    return res.status(httpStatus.INTERNAL_SERVER_ERROR).jsend.error(err);
+  }
+}
```
Comment (nchen000) [inline src/app/deleteItem/deleteItem.js]: `deleteItem.js` should be under `src/app/list`

File: src/app/deleteItem/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import Express from 'express';
+import Joi from '@hapi/joi';
+import deleteItem from './deleteItem';
+import validateRequest from '../../middleware/validateRequest';
+
+const router = Express.Router();
+const schema = Joi.object({
+  groceryID: Joi.number(),
```
Comment (nchen000) [inline src/app/deleteItem/index.js]: you can change `Joi.number()` to `Joi.number().integer().required()`. See [@hapi/joi docs](https://hapi.dev/family/joi/#example) for more examples.

File: src/app/deleteItem/index.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+});
+
+const deleteItemValidation = validateRequest(schema, 'body');
+
+router.put('/deleteItem', deleteItemValidation, deleteItem);
+
+export default router;
```
Comment (nchen000) [inline src/app/deleteItem/index.js]: `index.js` should be under `src/app/list`.

File: src/app/index.js [hunk 4]
```diff
@@ -5,6 +5,7 @@ import compression from 'compression';
 import jsend from 'jsend';
 import healthCheck from './healthCheck';
 import authentication from './authentication';
+import deleteItem from './deleteItem';
```
Comment (nchen000) [inline src/app/index.js]: change these two lines into `import list from './list'` and `app.use('/list', list);`

**Comment by shirleytang0121** [comment]
waiting for nan's PR merge in order to include idTokenValidation

File: src/app/deleteItem/_tests_/deleteItem.test.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import deleteItem from '../deleteItem';
```
Comment (nchen000) [inline src/app/deleteItem/_tests_/deleteItem.test.js]: `deleteItem.test.js` should be under `src/app/list/__tests__/` and the name of test directory should be `__tests__` (two underscores instead one).

File: src/app/list/index.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+const router = Express.Router();
+const schema = Joi.object({
+  groceryID: Joi.number().integer().required(),
+});
+
+const deleteItemValidation = validateRequest(schema, 'body');
+
+router.put('/deleteItem', asyncHandler(tokenAuthentication), deleteItemValidation, deleteItem);
+
```
Comment (nchen000) [inline src/app/list/index.js]: you can change `/deleteItem` to `/deleteitem` to keep our route naming consistent.

**Pull request merged**